### PR TITLE
Migrate to eslint-config-rainbow

### PR DESCRIPTION
**Pull request by bcomnes**
I was getting a ton of weird compatibility issues with tsdx, with resolution errors and whatnot.  Still looking into it, but I suspect some of them have to do with the use of the older eslint? Not sure yet.

I'll drop some comments on some of the rough edges.

### CODE REVIEW

File: .eslintrc.js
Comment (bcomnes) [inline .eslintrc.js:7]: eslint didn't like the alias stuff.

File: package.json
Comment (bcomnes) [inline package.json:17]: Switching directly to eslint since it seemed like there were some resolution errors when we ran listing through tsdx.

File: package.json
Comment (bcomnes) [inline package.json:66]: Not sure why but plugin-jest was not resolving correctly from rainbow or tsdx, so I added it top level.  First time seeing this issue, if there is something I can do in eslint-config-rainbow I'll add a fix there.

File: src/components/button/ConnectButtonV1.tsx
Comment (bcomnes) [inline src/components/button/ConnectButtonV1.tsx:2]: This was not resolving properly according to eslint, even when I added it to the package.json.

**Comment by bcomnes** [comment]
Oh wow, not getting those lining errors locally.  🤬

**Comment by bcomnes** [comment]
Oh derp, need to install deps on the website to lint correctly.

**Comment by bcomnes** [comment]
Actually don't merge this yet, I want to see if I can solve module resolution issues better in eslint-config-rainbow.

**Comment by bcomnes** [comment]
Ok, so I guess this shared-config doesn't play that nicely with tsdx unfortunately, and this is really the only approach is to do what we did here.

Open to ideas @estebanmino if you have any ideas on how to make eslint-config-rainbow work better with tsdx either in this repo or upstream, otherwise I would say this is good to go.

**Review by estebanmino — APPROVED** [review]
:shipit:

**Comment by bcomnes** [comment]
Merging

**Pull request merged**
